util.js

```diff
@@ -0,0 +1,13 @@
+define(
+  []
+, function() {
+
+    function isValidNumber(num) {
+      return (/^[+]?([0-9]+(?:[\.][0-9]*)?|\.[0-9]+)$/).test(num)
```

Comment:
Oh no, don't use regex for this, it's not a very good fit for the job. Here's a more solid function:

```javascript
function isValidTimeout(n) {
  return !isNaN(parseFloat(n)) && isFinite(n) && Math.max(0, n) === n
}
```

The first two clauses will check that `n` is a number, and the last will check that it's in the range of `0` and all finite numbers above.

Also, please don't make this a separate module, just put it in the middleware module, where it's in scope of the timeout middleware. There's no reason to make this a separate module when it's not reused anywhere else.

Reply:
Libraries, like `jQuery. d3` etc. doesn't force user and internally check whether the provided timeout is correct or not. If correct, then it will add the timeout. Else, it will set the default value.

Ref: 
[jquery-timeout!](https://github.com/jquery/jquery/blob/master/src/ajax.js#L687)
[d3-timeout!](https://github.com/d3/d3-request/blob/master/src/request.js#L112)

If agreed, then we can update the current code(as per d3, jQuery, like, `req.timeout > 0` and add the same check in middleware.js, instead of adding `isValidTimeout` method, so that if correct timeout is passed, we will use it. Else, will use our default value of `30 milliseconds`.

If not, I am happy to update codebase as per above requested changes. Also, please let me know the error which we need to send in case of invalid timeout. Can we use `"400": ["bad-request", "client-error"]`.

Thoughts??

Reply:
Ahh yes, that’s an elegant check, let’s do that – nice find!

As for the error: just throw, don’t send an error response.

Reply:
Updated the codebase as requested.

I have also added an error response `"400"` with `Invalid timeout` message because of the following reasons:
1. If we send `res(null)` without any error response, then code will break at different levels in address lib's like, in `logger` method and `is-view.js` etc..., where we are not handling an empty response.
2. By sending with `"400"`, we let user to know that its a `bad request` and message will tell it because of timeout.

Please let me know in case it needs further updation.